src/client.js

```diff
@@ -82,7 +83,7 @@ class MistralClient {
         'Authorization': `Bearer ${this.apiKey}`,
       },
       body: method !== 'get' ? JSON.stringify(request) : null,
-      timeout: this.timeout * 1000,
+      signal: signal ?? AbortSignal.timeout(this.timeout * 1000),
```

Comment:
Due to the limitations of what's possible when attempting to merge a user provided signal and a timeout signal, this simple approach may be the best that can be done without passing in an AbortController and implementing a timeout that calls its abort method. Thus if the user provides a signal, they are responsible for managing timeouts as well. This would be non-obvious to most users though. On the flip side, passing around an AbortController seems non-standard. 

We could document the parameter with something along these lines:
"Users must manage timeouts for their own AbortSignals. Without a user-provided signal, a default timeout of [x] seconds applies automatically."

Reply:
Actually, that may not be the case re: merging, as the signal.reason /is/ available and should be possible to transfer to a new merged signal.